Snowpark_For_Python_ML.ipynb

```diff
@@ -231,218 +226,89 @@
-    "    test_accuracy_threshold: float, \n",
-    "    save_model: bool) -> Variant:\n",
-    "    \n",
-    "    from sklearn.compose import ColumnTransformer\n",
-    "    from sklearn.pipeline import Pipeline\n",
-    "    from sklearn.preprocessing import PolynomialFeatures\n",
-    "    from sklearn.preprocessing import StandardScaler\n",
-    "    from sklearn.linear_model import LinearRegression\n",
-    "    from sklearn.model_selection import train_test_split, GridSearchCV\n",
+    "from snowflake.ml.modeling.compose import ColumnTransformer\n",
+    "from snowflake.ml.modeling.pipeline import Pipeline\n",
+    "from snowflake.ml.modeling.preprocessing import PolynomialFeatures, StandardScaler\n",
+    "from snowflake.ml.modeling.linear_model import LinearRegression\n",
+    "from snowflake.ml.modeling.model_selection import GridSearchCV\n",
     "\n",
-    "    import os\n",
-    "    from joblib import dump\n",
+    "CROSS_VALIDATION_FOLDS = 10\n",
+    "POLYNOMIAL_FEATURES_DEGREE = 2\n",
     "\n",
-    "    # Load features\n",
-    "    df = session.table(features_table).to_pandas()\n",
+    "# Create train and test snowpark dataframes\n",
+    "train_df, test_df = session.table(\"MARKETING_BUDGETS_FEATURES\").random_split(weights=[0.8, 0.2], seed=0)\n",
     "\n",
-    "    # Preprocess the Numeric columns\n",
-    "    # We apply PolynomialFeatures and StandardScaler preprocessing steps to the numeric columns\n",
-    "    # NOTE: High degrees can cause overfitting.\n",
-    "    numeric_features = ['SEARCH_ENGINE','SOCIAL_MEDIA','VIDEO','EMAIL']\n",
-    "    numeric_transformer = Pipeline(steps=[('poly',PolynomialFeatures(degree = polynomial_features_degrees)),('scaler', StandardScaler())])\n",
+    "# BUILD MODEL\n",
     "\n",
-    "    # Combine the preprocessed step together using the Column Transformer module\n",
-    "    preprocessor = ColumnTransformer(\n",
-    "        transformers=[\n",
-    "            ('num', numeric_transformer, numeric_features)])\n",
     "\n",
-    "    # The next step is the integrate the features we just preprocessed with our Machine Learning algorithm to enable us to build a model\n",
-    "    pipeline = Pipeline(steps=[('preprocessor', preprocessor),('classifier', LinearRegression())])\n",
-    "    parameteres = {}\n",
+    "# Preprocess the Numeric columns\n",
+    "# We apply PolynomialFeatures and StandardScaler preprocessing steps to the numeric columns\n",
+    "# NOTE: High degrees can cause overfitting.\n",
+    "numeric_features = ['SEARCH_ENGINE','SOCIAL_MEDIA','VIDEO','EMAIL']\n",
+    "numeric_transformer = Pipeline(steps=[('poly',PolynomialFeatures(degree = POLYNOMIAL_FEATURES_DEGREE)),('scaler', StandardScaler())])\n",
     "\n",
-    "    X = df.drop('REVENUE', axis = 1)\n",
-    "    y = df['REVENUE']\n",
+    "# Combine the preprocessed step together using the Column Transformer module\n",
+    "preprocessor = ColumnTransformer(\n",
```

Comment:
This is not a great example to display native pipeline capabilities. PolynomialFeatures is a transformer that produces indeterministic number of output columns (num of columns determined by input params and number of input features). In this particular example, with given params and 4 input features, it produces 256 output columns. Users would have to guess this number based on the given configuration and correctly wire columns between PolynomialFeatures, and StandardScaler. It would be easier to just use ColumnTransformer here.